File: support_tools/swigwin-3.0.12/Examples/test-suite/php/ignore_parameter_runme.php
<?php

require "tests.php";
require "ignore_parameter.php";

// New functions
check::functions(array(jaguar,lotus,tvr,ferrari,sportscars_daimler,sportscars_astonmartin,sportscars_bugatti,sportscars_lamborghini));
// New classes
check::classes(array(ignore_parameter,SportsCars,MiniCooper,MorrisMinor,FordAnglia,AustinAllegro));
// No new vars
check::globals(array());

check::equal(jaguar(2,3.4),"hello",'jaguar(2,3.4)=="hello"');
check::equal(lotus("eek",3.4),101,'lotus("eek",3.4)==101');
check::equal(tvr("eek",2),8.8,'tvr("eek",2)==8.8');
check::equal(ferrari(),101,'ferrari(2)==101');

$sc=new sportscars();
check::classname("sportscars",$sc);
check::equal($sc->daimler(2,3.4),"hello",'$sc->daimler(2,3.4)=="hello"');
check::equal($sc->astonmartin("eek",3.4),101,'$sc->mastonmartin("eek",3.4)==101');
check::equal($sc->bugatti("eek",2),8.8,'$sc->bugatti("eek",2)==8.8');
check::equal($sc->lamborghini(),101,'$sc->lamborghini(2)==101');

$mc=new minicooper(2,3.4);
check::classname("minicooper",$mc);

$mm=new morrisminor("eek",3.4);
check::classname("morrisminor",$mm);

$fa=new fordanglia("eek",2);
check::classname("fordanglia",$fa);

$aa=new austinallegro();
check::classname("austinallegro",$aa);

check::done();
?>

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/ignore_parameter_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'ignore_parameter'

include Ignore_parameter

# Global function tests
raise RuntimeError unless jaguar(0, 1.0) == "hello"
raise RuntimeError unless lotus("foo", 1.0) == 101
raise RuntimeError unless tvr("foo", 0) == 8.8
raise RuntimeError unless ferrari() == 101

# Member function tests
sc = SportsCars.new
raise RuntimeError unless sc.daimler(0, 1.0) == "hello"
raise RuntimeError unless sc.astonmartin("foo", 1.0) == 101
raise RuntimeError unless sc.bugatti("foo", 0) == 8.8
raise RuntimeError unless sc.lamborghini() == 101

# Constructor tests
MiniCooper.new(0, 1.0)
MorrisMinor.new("foo", 1.0)
FordAnglia.new("foo", 0)
AustinAllegro.new()

